src/ServiceControl/Infrastructure/Extensions/NegotiatorExtensions.cs

```diff
@@ -110,32 +111,32 @@ public static Negotiator WithPagingLinks(this Negotiator negotiator, int totalRe
 
             if (page != 1)
             {
-                AddLink(links, 1, "first", queryParams, url);
+                AddLink(links, 1, "first", queryParams, requestUri);
             }
 
             if (page > 1)
             {
-                AddLink(links, page - 1, "prev", queryParams, url);
+                AddLink(links, page - 1, "prev", queryParams, requestUri);
             }
 
             if (page != lastPage)
             {
-                AddLink(links, lastPage, "last", queryParams, url);
+                AddLink(links, lastPage, "last", queryParams, requestUri);
             }
 
             if (page < lastPage)
             {
-                AddLink(links, page + 1, "next", queryParams, url);
+                AddLink(links, page + 1, "next", queryParams, requestUri);
             }
 
             return negotiator.WithHeader("Link", String.Join(", ", links));
         }
 
-        static void AddLink(ICollection<string> links, int page, string rel, string queryParams, Url url)
+        static void AddLink(ICollection<string> links, int page, string rel, string queryParams, Uri uri)
         {
-            url.Query = queryParams + "page=" + page;
-
-            links.Add($"<{url}>; rel=\"{rel}\"");
+            var query = queryParams + "page=" + page;
+        
+            links.Add($"<{uri.AbsolutePath + query}>; rel=\"{rel}\"");
```

Comment:
I had to look up what `AbsolutePath` does https://docs.microsoft.com/en-us/dotnet/api/system.uri.absolutepath?view=netframework-4.7.2 because I was like "hu? we are trying to avoid using the absolute path"

Reply:
strictly speaking, we want the absolute path as it's not relative to the "current location". We want to omit the scheme + authority on the URI spec:
> URI = scheme:[//authority]path[?query][#fragment]

 which `AbsolutePath` does:

> The path information does not include the scheme, host name, or query portion of the URI.

Reply:
Yeah exactly, like I said I had to look up what it does because in my mind I thought we have to use relative path. Your code makes sense, I was just reflecting about my thought process publicly, ignore me 💃